lib/improver/utilities/cube_manipulation.py

```diff
@@ -685,3 +687,125 @@ def sort_coord_in_cube(cube, coord, order="ascending"):
+    all the dimension coordinates can be reordered by specifying the coordinate
+    names in the desired order.
+
+    Args:
+        cube (iris.cube.Cube):
+            Cube where the ordering will be enforced to match the order within
+            the coord_names.
+        coord_names (list or str):
+            List of the names of the coordinates to order. If a string is
+            passed in, only the single specified coordinate is reordered.
+        anchor (str):
+            String to define where within the range of possible dimensions
+            the specified coordinates should be located. This could be either:
+            "start" or "end".
+        promote_scalar (bool):
+            If True, any coordinates that are matched and are not dimension
+            coordinates are promoted to dimension coordinates.
+            If False, any coordinates that are matched and are not dimension
+            coordinates will not be considered in the reordering.
+        raise_exception (bool):
+            Option as to whether an exception should be raised, if the
+            requested coordinate is not present.
+
+    Returns:
+        cube (iris.cube.Cube):
+            Cube where the requirement for the dimensions to be in a particular
+            order will have been enforced.
+
+    Raises:
+        ValueError: The anchor argument must have a value of either start or
+            end.
+        CoordinateNotFoundError: The requested coordinate is not available on
+            the cube.
+        ValueError: Multiple coordinates match the partial name provided.
+
+    """
+    if isinstance(coord_names, unicode):
+        coord_names = str(coord_names)
+    if isinstance(coord_names, str):
+        coord_names = [coord_names]
+
+    if anchor not in ["start", "end"]:
+        msg = ("The value for the anchor must be either 'start' or 'end'."
+               "The value specified for the anchor was {}".format(anchor))
+        raise ValueError(msg)
+
+    # Determine coordinate indices for use in creating a dictionary.
+    # These indices are either relative to the start or end of the available
+    # dimension coordinates.
+    coord_indices = np.array(range(len(coord_names)))
+    if anchor == "end":
+        coord_indices = sorted(len(cube.dim_coords) - coord_indices)
+    coord_dict = dict(zip(coord_names, coord_indices))
+
+    for coord_name in coord_dict.keys():
+        # Deal with the coord_name being a partial match to the actual
+        # coordinate name.
+        if cube.coords(coord_name):
+            full_coord_name = coord_name
+        else:
+            coord = [c for c in cube.coords() if coord_name in c.name()]
+            # If the coordinate is desired, raise an exception
+            # if the coordinate is missing.
+            if len(coord) == 0:
+                if raise_exception:
+                    msg = ("The requested coordinate {} is not a coordinate "
```

Comment:
Might it be worth looking at whether or not promote_scalar has been set and whether the coordinate is a scalar or not. I can request a raise_exception behaviour, and yet if I don't promote_scalar, my scalar time coordinate is effectively unavailable for reordering; I might expect that to give me an exception. Thoughts?

Reply:
I've added the option to raise an exception if the coordinate isn't a dimension coordinate and promote_scalar hasn't been enabled, to avoid a silent behaviour in this case.